dask_ml/model_selection/_incremental.py

```diff
@@ -652,13 +665,17 @@ class IncrementalSearch(BaseIncrementalSearch):
         If ``patience`` is used the maximum number of ``partial_fit`` calls
         between ``score`` calls.
 
-    tol : float, default 0.001
+    tol : float, default -0.01
         The required level of improvement to consider stopping training on
         that model. The most recent score must be at at most ``tol`` better
         than the all of the previous ``patience`` scores for that model.
         Increasing ``tol`` will tend to reduce training time, at the cost
         of worse models.
 
+        Setting this value to be negative allows for some noise tolerance in
```

Comment:
I think this doesn't sit with the previous paragraph (maybe the previous paragraph is wrong).

It sounds like `tol` is influencing the question "should we stop?". Something like  `if current_score - previous_score > tol: stop`. Setting this to negative would mean we'll consider stopping on a model that performed worse than the previous one, while with positive the final model must always be better than the previous `patience` (until we hit `max_iter`).

I also can't tell what we mean by "more accurate estimator", or why that would be. I can imagine it being because we're not overfitting as much, but I can also imagine other explanations.

Looking at the code, we say `if all(current_score < old + tol for old in patience)`. So a negative tol makes that condition more likely to be true, i.e. we continue training even when our score has dipped. So I think the first paragraph explaining `tol` is incorrect?

Reply:
I set it negative to stop training when the scores starts to decrease. I'd expect to stop when performance starts to degrade, not when performance doesn't increase enough.

I guess my expectations are wrong; PyTorch's [`ReduceLROnPlateau`][1] performs it's action when `all(old > best + threshold for old in plateau)` for some positive `threshold`. That is, it stops increasing when models haven't improved enough in `patience` epochs.

TODO:

- [ ] default `tol=1e-4` (PyTorch's `ReduceLROnPlateau` `threshold`).

[1]:https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau